sunpy/timeseries/timeseriesbase.py

```diff
@@ -445,6 +445,26 @@ def concatenate(self, others, same_source=False, **kwargs):
         object._sanitize_units()
         return object
 
+    def reindex(self, index, **kwargs):
+        """
+        Returns a new time series with the chosen index.
+
+        Parameters
+        ----------
+        index : `~pandas.DatetimeIndex`
+            A valid index column.
+
+        Returns
+        -------
+        `~sunpy.timeseries.TimeSeries`
+            A `~sunpy.timeseries.TimeSeries` with new index.
+
+        Notes
+        -----
+        This method is a wrapper around `~pandas.DataFrame.reindex`
+        """
+        return GenericTimeSeries(self._data.reindex(index, **kwargs), units=self.units)
```

Comment:
Is the timeseries metadata taken account of?

Reply:
is copying metadata enough or should I modify it?

Reply:
Copying it should be enough? I'm not too familiar with time series metadata. I don't think there is a time value but I could be quite wrong.

Reply:
look like there is,
attached metadata for goes xrs test data:
```console
[(   <sunpy.time.timerange.TimeRange object at 0x7f36b841ef10>
      Start: 2011-06-06 23:59:59
      End:   2011-06-07 23:59:57
      Center:2011-06-07 11:59:58
      Duration:0.9999730324069096 days or
             23.99935277776583 hours or
             1439.9611666659498 minutes or
             86397.66999995698 seconds,
  ['xrsa', 'xrsb'],
  MetaDict([('simple', True),
            ('bitpix', 8),
            ('naxis', 0),
            ('extend', True),
            ('date', '26/06/2012'),
            ('numext', 3),
            ('telescop', 'GOES 15'),
            ('instrume', 'X-ray Detector'),
            ('object', 'Sun'),
            ('origin', 'SDAC/GSFC'),
            ('date-obs', '07/06/2011'),
            ('time-obs', '00:00:00.000'),
            ('date-end', '07/06/2011'),
            ('time-end', '23:59:57.632'),
            ('comment',
             'Energy band information given in extension 1Fluxes information given in extension 2Status word information given in extension 3Ref.: Solar X-Ray Measurements from SMS-1, SMS-2, and GOES-1;Information for Data Users. Donnelly et al,  June 1977.NOAA TM ERL SEL-48Ref.: SMS GOES Space Environment Monitor Subsystem,Grubb, Dec 75, NOAA, Technical Memorandum ERL SEL-42.Ref.:Expresions to Determine Temperatures and EmissionMeasures for Solar X-ray events from GOES Measurements.Thomas et al, 1985, Solar Physics 95, pp 323-329.'),
            ('history', ''),
            ('keycomments',
             {'SIMPLE': 'Written by IDL:  Tue Jun 26 09:32:13 2012',
              'EXTEND': 'File contains extensions',
              'DATE': 'file creation date',
              'NUMEXT': 'no. of extensions in file',
              'TELESCOP': 'spacecraft',
              'ORIGIN': 'written by Solar DAC at GSFC',
              'DATE-OBS': 'date of first obs.(DD/MM/YYYY)',
              'TIME-OBS': 'UT time of first obs.(HH:MM:SS.XXX)',
              'DATE-END': 'date of last obs.(DD/MM/YYYY)',
              'TIME-END': 'UT time of last obs.(HH:MM:SS.XXX)'})]))]
```

Reply:
Copy it and then others can chime in if anything needs updating after.

Reply:
time range issue should be fixed now👍